ansible_base/oauth2_provider/checks/permisssions_check.py

```diff
@@ -0,0 +1,154 @@
+import inspect
+from typing import Optional, Type, Union
+
+from django.apps import AppConfig
+from django.conf import settings
+from django.core.checks import CheckMessage, Debug, Error, Warning
+from django_extensions.management.commands.show_urls import Command as ShowUrlsCommand
+from rest_framework.permissions import BasePermission, OperandHolder, OperationHolderMixin, SingleOperandHolder
+from rest_framework.views import APIView, View
+
+from ansible_base.oauth2_provider.permissions import OAuth2ScopePermission
+
+# Fallback value in the event that ANSIBLE_BASE_OAUTH2_PROVIDER_PERMISSIONS_CHECK_IGNORED_VIEWS setting is undefined
+FALLBACK_OAUTH2_SCOPE_PERMISSION_CHECK_IGNORE_LIST = set(
+    [
+        "drf_spectacular.views.SpectacularAPIView",
+        "drf_spectacular.views.SpectacularRedocView",
+        "drf_spectacular.views.SpectacularSwaggerView",
+    ]
+)
```

Comment:
Can probably be done away with in favor of a dynamic setting for oauth2_provider.